5-Heroku/server/todo_service.js

```diff
@@ -0,0 +1,94 @@
+const mongoose = require("mongoose");
+const uuid = require("uuid");
+require('dotenv').config();
+
+const Schema = mongoose.Schema;
+
+const MONGO_URI = process.env.MONGO_URI;
+
+mongoose.connect(MONGO_URI, {
+  useNewUrlParser: true,
+  useUnifiedTopology: true,
+});
+
+mongoose.connection.on("connected", () => {
+  console.log("MobgoDB is connected!");
+});
+
+const TodoSchema = new Schema({
+  user_id: String,
+  id: String,
+  text: String,
+  check: Boolean,
+});
+
+const Todo = mongoose.model("Todo", TodoSchema);
+
+function dbUpdateCallback(req, res, successStatus) {
+  return function (error, doc) {
+    if (error) {
+      console.log(error);
+      res.status(500).send("Database error!");
+    } else {
+      res.sendStatus(successStatus);
+    }
+  };
+}
+
+module.exports.getItems = (req, res) => {
+  let userId;
+
+  if (req.cookies === undefined ||
```

Comment:
u have an assumption here that `getItems` is the first thing that happens and so if the user doesn't have a cookie it will create it here.

this is what known to be `implicit` we always want to be `explicit`.

what I want you to do is to write a middleware that for each request to the server it will run this logic. it will later help us in authorization as well